### ISSUE TITLE
Best practices for portable disk format

### ISSUE THREAD

**Issue by core-man** [seismo-learn/seismology101#147]
As we may have different OS, we may choose a file system for our portable disk so that it can be used in different OS.

**Comment by seisman** [comment]
I'm afraid it's impossible.

**Comment by core-man** [comment]
> I'm afraid it's impossible.

at least two of the three OS.

exFAT can be uses in those three OS.

**Comment by seisman** [comment]
It's fragile and you may lose your data. See the comments in https://www.ruanyifeng.com/blog/2018/10/exfat.html

**Comment by core-man** [comment]
> It's fragile and you may lose your data. See the comments in https://www.ruanyifeng.com/blog/2018/10/exfat.html

Yes, that is the reason why we need to prepare this pratice so that new babies know it. They will face this issue when they buy one portable disk.

I think exFat is good to store seismic waveform data, while we need another format to store files with codes.

see https://core-man.github.io/blog/post/backup/#%E7%A1%AC%E7%9B%98%E5%A4%87%E4%BB%BD

**Comment by seisman** [comment]
> I think exFat is good to store seismic waveform data



> I think exFat is good to store seismic waveform data

As mentioned above, you will take the risk losing your data, not only the file permissions.

**Comment by core-man** [comment]
I mean we have to choose different format for different purposes and also for different OS we have. Of course, there are no formats that can be used for all purposes and OS.

**Comment by core-man** [comment]
> > I think exFat is good to store seismic waveform data
> 
> > I think exFat is good to store seismic waveform data
> 
> As mentioned above, you will take the risk losing your data, not only the file permissions.

Then we may better add a warnings.

**Comment by core-man** [comment]
> > I think exFat is good to store seismic waveform data
> 
> > I think exFat is good to store seismic waveform data
> 
> As mentioned above, you will take the risk losing your data, not only the file permissions.

BTW, a lot of seismic data at NTU are stored in NTFS or exFAT.

**Comment by seisman** [comment]
NTFS is good but exfat isn't.

**Comment by core-man** [comment]
> NTFS is good but exfat isn't.

But exFAT is now the default format for some portable disk.

**Comment by seisman** [comment]
If someone only uses one operating system, I think it's straightforward to choose the best format:

- macOS: APFS
- Windows: NTFS
- Linux: ext4, xfs or any others (these Linux-specific formats are supported by all Linux distros)

If someone uses two operating system, and want to share files between them, it's not easy to choose the best format, because all formats have their own limitations. 

Perhaps the best way is having a hard drive and a USB stick:

- the hard drives are used to store seismic data, and the disk format depends on the main OS you're using
- the USB stick can use the exFAT format, and is used to transfer files between OSs
